### PR TITLE
Improve fill-column module for Emacs 27+

### PR DESCRIPTION
**Pull request by rushsteve1**
Emacs 27 introduced `display-fill-column-indicator-mode` which replaces `hl-fill-column`. This commit adds some conditionals to use `display-fill-column-indicator-mode` on Emacs 27+, or falls back to `hl-fill-column-mode` on older versions.

This also fixes the deprecation message so I removed it. Once Emacs 27 is more common I would suggest marking it as deprecated again and removing the module entirely. I'll add the message back if you still want it there.

I suggest merging #3815 first since that adds documentation which I can update to match this change.

Any and all suggestions welcome! I'm not great at Elisp yet and in particular the `add-hook!` macro gave me trouble which led to a bit of repetition in the code.

### CODE REVIEW

**Comment by rushsteve1** [comment]
Two additional commits with changes suggested on Discord.

File: modules/ui/fill-column/autoload.el [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 ;;;###autoload
-(add-hook! '(text-mode-hook prog-mode-hook conf-mode-hook)
-           #'hl-fill-column-mode)
+;; Emacs 27 introduced `display-fill-column-indicator-mode' which should be
+;; used instead of `hl-fill-column-mode'
+(if EMACS27+
+    (add-hook! '(text-mode-hook prog-mode-hook conf-mode-hook)
+               #'display-fill-column-indicator-mode)
+    (add-hook! '(text-mode-hook prog-mode-hook conf-mode-hook)
+               #'hl-fill-column-mode))
```
Comment (hlissner) [inline modules/ui/fill-column/autoload.el:12]: Could you replace this duplication with a delegate? e.g.

```elisp
;;;###autoload
(defun +fill-column-enable-h (&optional arg)
  (interactive "p")
  (if (fboundp 'display-fill-column-indicator-mode)
      (display-fill-column-indicator-mode arg)
    (hl-fill-column-mode arg)))
```

Then we can have only one `add-hook!`, and one `:when` check in config/default/+{evil,emacs}-bindings.el.

**Review by hlissner — CHANGES_REQUESTED** [review]

File: modules/ui/fill-column/autoload.el (hunk 1 above)
Comment (rushsteve1) [reply modules/ui/fill-column/autoload.el]: Ah, I spent forever trying to reduce that duplication. Good to know that trick, should help me a lot.

**Comment by rushsteve1** [comment]
Thanks for the suggestions @hlissner, I made them and pushed the changes.

Also ignore the force-push, I had some weird merge commits I needed to remove.

**Comment by hlissner** [comment]
Looks good to me. Thanks!

**Pull request merged**